pkg/asset/installconfig/gcp/validation.go

```diff
@@ -112,32 +112,71 @@ func validateServiceAccountPresent(client API, ic *types.InstallConfig) field.Er
 	return allErrs
 }
 
+// FIXME: how to keep in sync with pkg/asset/machine/worker.go?
```

Comment:
From what I can see the, the default machine pool is still using amd64 instances:

https://github.com/openshift/installer/blob/c2d7db9d4eedf7b79fcf975f3cbd8042542982ca/pkg/asset/machines/worker.go#L127-L135

& 

https://github.com/openshift/installer/blob/c2d7db9d4eedf7b79fcf975f3cbd8042542982ca/pkg/asset/machines/worker.go#L500-L505

When setting the default machine pool, it should make use of `defaultInstanceTypeForArch`.

To keep in sync, you could export `defaultGCPMachinePoolPlatform` and make use of it in the validation.

Reply:
`arm64` defaults will be set by https://github.com/openshift/installer/pull/7258.

Reply:
@patrickdillon it looks like I cannot import `github.com/openshift/installer/pkg/asset/machines` because it causes a cycle dep:
```bash
package github.com/openshift/installer/cmd/openshift-install
	imports github.com/openshift/installer/cmd/openshift-install/agent
	imports github.com/openshift/installer/pkg/agent
	imports github.com/openshift/installer/pkg/asset/agent/image
	imports github.com/openshift/installer/pkg/asset/agent
	imports github.com/openshift/installer/pkg/asset/installconfig
	imports github.com/openshift/installer/pkg/asset/installconfig/gcp
	imports github.com/openshift/installer/pkg/asset/machines
	imports github.com/openshift/installer/pkg/asset/ignition/machine
	imports github.com/openshift/installer/pkg/asset/installconfig: import cycle not allowed
package github.com/openshift/installer/cmd/openshift-install
	imports github.com/openshift/installer/cmd/openshift-install/agent
	imports github.com/openshift/installer/pkg/agent
	imports github.com/openshift/installer/pkg/asset/agent/image
	imports github.com/openshift/installer/pkg/asset/agent
	imports github.com/openshift/installer/pkg/asset/installconfig
	imports github.com/openshift/installer/pkg/asset/installconfig/gcp
	imports github.com/openshift/installer/pkg/asset/machines
	imports github.com/openshift/installer/pkg/asset/machines/gcp
	imports github.com/openshift/installer/pkg/asset/installconfig/gcp: import cycle not allowed
```

Reply:
I have inverted the logic where the default lives to avoid the cycle dep.